packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -166,79 +240,77 @@ export const ItemDetailsModal: React.FC<ItemDetailsModalProps> = ({
     onReset();
   };
 
-  const checkAllocatedQuantities = () => {
-    const values = getValues();
-    const allocatedUnits = batchRows.reduce(
-      (total, batch) =>
-        (total += Number(values[batch.id] || 0) * batch.packSize),
-      Number(values['placeholder'] || 0)
-    );
-
-    setAllocated(Math.floor(allocatedUnits / packSize));
-  };
-
-  const allocateQuantities = () => {
+  const allocateQuantities = (
+    newValue: number,
+    issuePackSize: number | null
+  ) => {
+    setValue('quantity', String(newValue));
     // if invalid quantity entered, don't allocate
-    if (quantity < 1 || Number.isNaN(quantity) || !summaryItem) {
+    if (newValue < 1 || Number.isNaN(newValue)) {
```

Comment:
Changes here because I am bad with numbers and it took me a bit to make sure I understand, but should be the same as what you had except assigning to a `"batchRow"` rather than using `setValue`